### DISCUSSION TITLE
Proper Examples

### DISCUSSION THREAD

**Discussion by naingminoo99** [awslabs/aws-sdk-swift#1013] [Q&A] [answered]
I am currently using the aws-sdk-ios and considering a transition to the Swift SDK since I will be utilizing it with SwiftUI. However, I'm facing challenges finding comprehensive documentation or example code on how to establish an AWS connection, similar to what was available in the previous SDK. I believe that providing a proper example project with clear implementation steps would greatly benefit many developers in understanding and using the Swift SDK effectively.

**Comment by jbelkins** [comment]
**The AWS SDK for Swift is currently in Developer Preview status.**

As such, the SDK is for evaluation & feedback purposes only.  AWS does not recommend using this SDK for any other purpose at this time, and we do not have a timeframe for when it will be officially released as a generally available product, i.e. "1.0" status.

In the interim, expect that our documentation may be incomplete, missing, or in a draft state.  While we will improve and expand it as we approach general availability, we do not have a timeframe for when any part of our documentation will be published.

I recommend that you continue to use `aws-sdk-ios` for the foreseeable future if you are building or maintaining a production product.  When this SDK becomes generally available, we will make it known to the developer community.

**Comment by JetForMe** [comment]
Thing is, it's virtually impossible to evaluate and give feedback. This API is incredibly obtuse to look at, and without a basic example (say, how do I connect to an S3 bucket, given the key pair I need for access), we can't give you any meaningful feedback.

**Reply by JeanPierreFig** [reply]
This is so frustrating. I have been able to get it to work using env variable in Xcode but if I want to risk it and ship it. I can't seem to find the correct way to ship it with the keys.

**Reply by sichanyoo** [reply]
@JeanPierreFig 

You should _**not**_ ship your application with your access & secret key. Instead, recommended approach for giving access to AWS resources for users using your application is to use AWS Cognito. [Here](https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html)'s the official doc.

**Reply by sholtomaud** [reply]
Production ready secure examples please. 
Also provide the production ready DevSecOps CodePipeline that will do all the testing of auth with Swift & Cognito Apple apps.

**Reply by dayaffe** [reply]
Our official developer guide complete with code examples can be found at the following locations:

Full PDF: https://docs.aws.amazon.com/pdfs/sdk-for-swift/latest/developer-guide/aws-sdk-swift-dg-v1.pdf#home
Site: https://docs.aws.amazon.com/sdk-for-swift/latest/developer-guide/home.html

**Comment by jbelkins** [comment, marked as answer]
Our official developer guide complete with code examples can be found at the following locations:

Full PDF: https://docs.aws.amazon.com/pdfs/sdk-for-swift/latest/developer-guide/aws-sdk-swift-dg-v1.pdf#home
Site: https://docs.aws.amazon.com/sdk-for-swift/latest/developer-guide/home.html
